src/effects/presets/effectchainpresetmanager.cpp

```diff
@@ -896,6 +931,10 @@ void EffectChainPresetManager::saveEffectsXml(QDomDocument* pDoc, const EffectsX
     QDomElement chainsElement = pDoc->createElement(EffectXml::kChainsRoot);
     rackElement.appendChild(chainsElement);
     for (const auto& pPreset : std::as_const(data.standardEffectChainPresets)) {
+        // Don't store the name '---', see readEffectsXml() for explanation.
+        if (pPreset->name() == kNoEffectString) {
```

Comment:
if we don't want to store the noeffect, shouldn't we skip it (continue) instead of setting the name empty and adding it?

Reply:
(as you did below)

Reply:
I added this only as safeguard to not dump loaded effects in the (unexpected) case that there are effects loaded despite the '---' preset name.
I can turn it into a VERIFY_OR_DEBUG_ASSERT.

Reply:
Fixed by the last commit https://github.com/mixxxdj/mixxx/pull/12561/commits/d9b104bee7a6a61909e81f618a5367ac281693d0